app/code/Magento/Inventory/Plugin/SourceTypeAttribute.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Plugin;
+
+use Magento\InventoryApi\Api\SourceRepositoryInterface;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\Inventory\Model\SourceTypeLinkManagement;
+
+class SourceTypeAttribute
```

Comment:
* Would be best to split the plugin into two plugins, each with single responsibility of manipulating extension attributes
* Please add descriptions to all classes